### PR TITLE
Int: Make _runtime and _client_ to only use public API of _shared

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
# The problem

_Please describe the motivation for this PR. Bonus points for code examples, error messages, and screenshots._

# This PR's solution

_Please describe the PR's content. Bonus points for code examples, error messages, screenshots._

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: src/orquestra/sdk/_shared/__init__.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from ._graphs import iter_invocations_topologically
+from ._logs import (
+    LogAccumulator,
+    LogOutput,
+    LogReader,
+    LogStreamType,
+    WorkflowLogs,
+    is_env_setup,
+    is_worker,
+)
```
Comment (alexjuda) [inline src/orquestra/sdk/_shared/__init__.py:10]: WDYT about leaving these inside a to-be-public `_shared.log` package? The names `is_env_setup` and `is_worker` make much more sense when they're "namespaced" by the `logs` module.

File: src/orquestra/sdk/_shared/_logs/__init__.py
Comment (alexjuda) [inline src/orquestra/sdk/_shared/_logs/__init__.py:1]: `_logs/__init__.py` -> `logs/__init__.py`? 🙏

File: src/orquestra/sdk/_shared/schema/workflow_run.py [hunk 2]
```diff
@@ -11,7 +11,8 @@
 import typing as t
 import warnings
 
-from .._dates import Instant
+from orquestra.sdk._shared.dates._dates import Instant
```
Comment (alexjuda) [inline src/orquestra/sdk/_shared/schema/workflow_run.py:14]: **nitpick**: any particular reason to switch from a relative import to an absolute, and (overly) nested import?

File: src/orquestra/sdk/_client/_base/_testing/_example_wfs.py [hunk 3]
```diff
@@ -324,9 +324,9 @@ def wf_with_log(msg: str):
 
 @sdk.task
 def get_exec_ctx() -> str:
-    import orquestra.sdk._shared._exec_ctx
+    import orquestra.sdk._shared.exec_ctx._exec_ctx
```
Comment (alexjuda) [inline src/orquestra/sdk/_client/_base/_testing/_example_wfs.py:327]: **issue**: private attribute access across sub-packages

**Review by alexjuda — CHANGES_REQUESTED** [review]
One "illegal" import found! (Illegal by our recent definition of what we don't want to do across sub-packages)

File: src/orquestra/sdk/_shared/__init__.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+from ._regex import SEMVER_REGEX, VERSION_REGEX
+from ._spaces import Project, ProjectRef, Workspace
+from ._storage import BaseModel, TypeAdapter
```
Comment (jamesclark-Zapata) [inline src/orquestra/sdk/_shared/__init__.py:15]: **nitpick (non-blocking):** ‏I would rather not export these Pydantic backwards compat classes from the root.

Comment (alexjuda) [reply src/orquestra/sdk/_shared/__init__.py:15]: Oh, good point! Any suggestions for the module name?

File: src/orquestra/sdk/_shared/__init__.py (hunk 1 above)
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_shared/__init__.py]: Good idea, Ill just remove it from top-level init in shared

File: src/orquestra/sdk/_shared/__init__.py (hunk 4 above)
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_shared/__init__.py]: Maybe the "orqdantic" package? WDYT? :D

File: src/orquestra/sdk/_shared/__init__.py [hunk 5]
```diff
@@ -1,21 +1,10 @@
...
-    LogStreamType,
-    WorkflowLogs,
-    is_env_setup,
-    is_worker,
-)
 from ._regex import SEMVER_REGEX, VERSION_REGEX
 from ._spaces import Project, ProjectRef, Workspace
-from ._storage import BaseModel, TypeAdapter
+from .logs import LogAccumulator, LogOutput, LogReader, LogStreamType, WorkflowLogs
+from .schema import result_is_packed
```
Comment (alexjuda) [inline src/orquestra/sdk/_shared/__init__.py:5]: Given we're having these to-be-public modules now, any reason to re-export it under the base init?
```suggestion
```

File: src/orquestra/sdk/_shared/abc.py [hunk 6]
```diff
@@ -15,8 +15,8 @@
 
 from orquestra.sdk._shared.exceptions import WorkspacesNotSupportedError
 
-from ._logs._interfaces import LogOutput, LogReader, WorkflowLogs
 from ._spaces._structs import Project, ProjectRef, Workspace
+from .logs._interfaces import LogOutput, LogReader, WorkflowLogs
```
Comment (alexjuda) [inline src/orquestra/sdk/_shared/abc.py:19]: **nitpick**: unless we have an `__all__` attribute in this module (`_shared.abc`), this line will re-export `LogOutput, LogReader, WorkflowLogs` under the public module `_shared.abc`, which might not be the intention. Unfortunately the same applies to all the public modules under `_shared.schema` which we might wanna keep out of scope of this ticket.

**Review by alexjuda — APPROVED** [review]
No blocking issues. LGTM. Please write the PR description before merging to main!

**Pull request merged**